From: Anchal Agarwal <anchalag@amazon.com>
Date: Tue, 27 Mar 2018 17:23:50 +0000
Subject: Revert "xen: dont fiddle with event channel masking in
 suspend/resume"

This reverts commit e91b2b1194335ca83d8a40fa4e0efd480bf2babe.
evtchn are supposed to be masked during resume however they are not
which causes special interrupts like PV spinlock to cause kernel
BUG() as its expects the IRQ to be masked. This causes instances
that are live migrated successfully to crash after few minutes.

Signed-off--by: Anchal Agarwal <anchalag@amazon.com>
Signed-off--by: Eduardo Valentin <eduval@amazon.com>
Reviewed-by: Frank van der Linden <fllinden@amazon.com>
Reviewed-by: Alakesh Haloi <alakeshh@amazon.com>
Reviewed-by: Vallish Vaidyeshwara <vallish@amazon.com>
Signed-off-by: Samuel Mendoza-Jonas <samjonas@amazon.com>
---
 drivers/xen/events/events_base.c | 13 ++++++++++---
 1 file changed, 10 insertions(+), 3 deletions(-)

diff --git a/drivers/xen/events/events_base.c b/drivers/xen/events/events_base.c
--- a/drivers/xen/events/events_base.c
+++ b/drivers/xen/events/events_base.c
@@ -542,6 +542,14 @@ static void bind_evtchn_to_cpu(evtchn_port_t evtchn, unsigned int cpu,
 	channels_on_cpu_inc(info);
 }
 
+static void xen_evtchn_mask_all(void)
+{
+	evtchn_port_t evtchn;
+
+	for (evtchn = 0; evtchn < xen_evtchn_nr_channels(); evtchn++)
+		mask_evtchn(evtchn);
+}
+
 /**
  * notify_remote_via_irq - send event to remote end of event channel via irq
  * @irq: irq of event channel to send event to
@@ -2106,6 +2114,7 @@ void xen_irq_resume(void)
 	struct irq_info *info;
 
 	/* New event-channel space is not 'live' yet. */
+	xen_evtchn_mask_all();
 	xen_evtchn_resume();
 
 	/* No IRQ <-> event-channel mappings. */
@@ -2296,7 +2305,6 @@ static int xen_evtchn_cpu_dead(unsigned int cpu)
 void __init xen_init_IRQ(void)
 {
 	int ret = -EINVAL;
-	evtchn_port_t evtchn;
 
 	if (xen_fifo_events)
 		ret = xen_evtchn_fifo_init();
@@ -2316,8 +2324,7 @@ void __init xen_init_IRQ(void)
 	BUG_ON(!evtchn_to_irq);
 
 	/* No event channels are 'live' right now. */
-	for (evtchn = 0; evtchn < xen_evtchn_nr_channels(); evtchn++)
-		mask_evtchn(evtchn);
+	xen_evtchn_mask_all();
 
 	pirq_needs_eoi = pirq_needs_eoi_flag;
 

